### ISSUE TITLE
GcpPluginSpec should use kfdef v1beta1

### ISSUE THREAD

**Issue by kunmingg** [kubeflow/kubeflow#4393]
/kind bug

GcpPluginSpec should use kfdef v1beta1.

GcpPluginSpec:
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/pkg/kfapp/gcp/gcp_plugin.go#L4
Config file:
https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_gcp_iap.yaml#L366

Other PluginSpec is having the same problem, we can handle in separate issue.

**Comment by jlewi** [comment]
It looks like the Auth field is different?
https://github.com/kubeflow/kubeflow/blob/2ae25c46ba0861dda683d92eda11fd8a588442de/bootstrap/pkg/kfapp/gcp/gcp_plugin.go#L9

/cc @gabrielwen

**Comment by jlewi** [comment]
I lean towards going with whatever is in the YAML files and either cleaning it up later if possible or just continuing to support what we have.

**Comment by jlewi** [comment]
@kunmingg Where did we land on this? Did we agree that this isn't blocking 0.7.0?

**Comment by kunmingg** [comment]
We can address it in next minor release or 1.0

**Comment by kunmingg** [comment]
reopen because it blocks basic auth v0.7.0

**Comment by kunmingg** [comment]
Error: secret source not found

How to reproduce:
```
kfctl build -V -f https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_gcp_basic_auth.0.7.0.yaml
../bin/kfctl apply -f
```

The root cause is:
`HashedSource` is deleted in v1beta1 kfdef, so 
`https://github.com/kubeflow/kubeflow/blob/master/bootstrap/pkg/apis/apps/configconverters/converters.go#L191` will drop `HashedSource` key when write GCP config (using v1alpha1) to config file.

**Comment by jlewi** [comment]
Basic auth is fixed by #4444

@kunmingg is there more work to be done here?

**Comment by gabrielwen** [comment]
Closing this issue as basic auth was fixed.
